File: app/src/main/java/ru/job4j/vkfriendskt/di/AppComponent.java
package ru.job4j.vkfriendskt.di;



import javax.inject.Singleton;

import dagger.Component;
import ru.job4j.vkfriendskt.viewmodels.FriendsListViewModel;
import ru.job4j.vkfriendskt.viewmodels.UserDetailsViewModel;

@Singleton
@Component(modules = {NetworkServiceModule.class})
public interface AppComponent {
    void injectTo(FriendsListViewModel friendsListViewModel);
    void injectTo(UserDetailsViewModel userDetailsViewModel);




}

File: app/src/main/java/ru/job4j/vkfriendskt/di/AppComponent.kt
package ru.job4j.vkfriendskt.di

import dagger.Component
import ru.job4j.vkfriendskt.viewmodels.FriendsListViewModel
import ru.job4j.vkfriendskt.viewmodels.UserDetailsViewModel
import javax.inject.Singleton

@Singleton
@Component(modules = [NetworkServiceModule::class])
interface AppComponent {
    fun injectTo(friendsListViewModel: FriendsListViewModel?)
    fun injectTo(userDetailsViewModel: UserDetailsViewModel?)
}